src/Simplify.elm

```diff
@@ -10359,26 +10352,52 @@ Use `returnsArgError` with the given last arg as `arg` when the last arg is alre
 alwaysReturnsLastArgError :
     String
     ->
-        { lastArgRepresents : String
-        , lastArg : Maybe (Node Expression)
+        { remainingArgCount : Int
+        , remainingArgs : List (Node Expression)
+        , lastArgRepresents : String
         }
     -> QualifyResources { a | fnRange : Range, parentRange : Range }
     -> Error {}
 alwaysReturnsLastArgError usingSpecificSituation config resources =
-    case config.lastArg of
+    case List.drop (config.remainingArgCount - 1) config.remainingArgs |> List.head of
+        Just lastArg ->
+            returnsArgError usingSpecificSituation { arg = lastArg, argRepresents = config.lastArgRepresents } resources
+
         Nothing ->
+            -- Not enough arguments
+            let
+                replacement : { description : String, fix : List Fix }
+                replacement =
+                    case config.remainingArgCount - List.length config.remainingArgs of
+                        1 ->
+                            { description = "identity"
+                            , fix =
+                                [ Fix.replaceRangeBy resources.parentRange
+                                    (qualifiedToString (qualify ( [ "Basics" ], "identity" ) resources))
+                                ]
+                            }
+
+                        2 ->
+                            { description = "always"
+                            , fix =
+                                [ Fix.replaceRangeBy resources.parentRange
+                                    (qualifiedToString (qualify ( [ "Basics" ], "always" ) resources))
+                                ]
+                            }
+
+                        _ ->
+                            -- Use-case is absent for now
+                            { description = "the " ++ config.lastArgRepresents ++ " argument"
+                            , fix = []
```

Comment:
should be `"(\\ " ++ repeat remainingArgCount "_ " ++ " -> identity)`

Reply:
Should be "(\\ " ++ repeat (remainingArgCount - 1) "_ " ++ " -> identity)` in that case I think.

- No missing argument: take the last argument
- 1 missing argument: `identity`
- 2 missing arguments: `always`
- 3 missing arguments: `\_ _ -> identity`?

Feels a bit inconsistent, no?